tasks/eva_2357/populate_ids.py

```diff
@@ -0,0 +1,196 @@
+import hashlib
+import argparse
+import pymongo
+import traceback
+import logging
+from ebi_eva_common_pyutils.config_utils import get_mongo_uri_for_eva_profile
+
+
+def generate_update_statement(hash_to_variant_ids, hash_to_accession_info):
+    variant_to_ids = {}
+    update_statements = []
+    for sve_hash, accessions in hash_to_accession_info.items():
+        ss_accession = f"ss{accessions.get('ss')}"
+        rs_accession = f"rs{accessions.get('rs')}" if accessions.get('rs') else accessions.get('rs')
+        variant_id = hash_to_variant_ids.get(sve_hash)
+
+        # Group all the ids (ss and rs) by variant id
+        if variant_to_ids.get(variant_id):
+            variant_to_ids[variant_id].add(ss_accession)
+            if rs_accession:
+                variant_to_ids[variant_id].add(rs_accession)
+        else:
+            ids = set()
+            ids.add(ss_accession)
+            if rs_accession:
+                ids.add(rs_accession)
+            variant_to_ids[variant_id] = ids
+
+    # Generate one update statement per variant_id
+    for variant_id, ids in variant_to_ids.items():
+        query = {"_id": variant_id}
+        update = {"$addToSet": {"ids": {"$each": list(ids)}}}
+        update_statements.append(pymongo.UpdateOne(query, update))
+    return update_statements
+
+
+def get_from_accessioning_db(mongo_handle, mongo_accession_db, sve_hashes):
+    # Get SS ID from accessioning DB
+    sve_collection = mongo_handle[mongo_accession_db]['submittedVariantEntity']
+    sve_filter = {"_id": {"$in": list(sve_hashes)}}
+    sve_projection = {"_id": 1, "accession": 1, "rs": 1}
+    cursor_accessioning = sve_collection.find(sve_filter, projection=sve_projection)
+    hash_to_accession_info = {}
+    for sve in cursor_accessioning:
+        sve_hash = sve.get("_id")
+        ss_accession = sve.get('accession')
+        rs_accession = sve.get('rs')
+        hash_to_accession_info[sve_hash] = {"ss": ss_accession}
+        # Get RS ID is variant is clustered
+        if rs_accession:
+            hash_to_accession_info[sve_hash]['rs'] = rs_accession
+    return hash_to_accession_info
+
+
+def get_variants_from_variant_warehouse(variants_collection):
+    projection = {"_id": 1, "files.sid": 1, "chr": 1, "start": 1, "ref": 1, "alt": 1, "type": 1}
+    return variants_collection.find({}, projection=projection)
```

Comment:
```suggestion
    return variants_collection.find({}, projection=projection, no_cursor_timeout=True)
```

Reply:
This. query could potentially return a large number of variant. Is it worth batching them  with `.limit()` and `.skip()` 
Maybe a question for @sundarvenkata-EBI

Reply:
Yes, this should be batched. I have noted it in another comment.